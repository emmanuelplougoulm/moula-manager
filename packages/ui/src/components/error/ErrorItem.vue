<template>
  <span v-if="text" id="error-item" data-testid="error-item" class="text-small-regular">
    {{ props.text }}
  </span>
</template>

<script setup lang="ts">
const props = defineProps({
  text: {
    type: String,
    default: ''
  }
});
</script>

<style scoped>
span {
  color: var(--error-30);
}
</style>
