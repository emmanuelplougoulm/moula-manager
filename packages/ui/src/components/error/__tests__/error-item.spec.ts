import { mount } from '@vue/test-utils';
import ErrorItem from '../ErrorItem.vue';

describe('ErrorItem tests', () => {
  it('should display the error message', () => {
    const wrapper = mount(ErrorItem, {
      props: {
        text: 'This field is required'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not display the error message', () => {
    const wrapper = mount(ErrorItem, {
      props: {
        text: undefined
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
