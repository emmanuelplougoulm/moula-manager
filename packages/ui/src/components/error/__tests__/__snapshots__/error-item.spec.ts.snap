// Vitest Snapshot v1

exports[`ErrorItem tests > should display the error message 1`] = `"<span data-v-c1d8433f=\\"\\" id=\\"error-item\\" data-testid=\\"error-item\\" class=\\"text-small-regular\\">This field is required</span>"`;

exports[`ErrorItem tests > should not display the error message 1`] = `"<!--v-if-->"`;
